### PR TITLE
internal: Add basic simnet xmr swap.

### LINKED ISSUE #2754: dex/networks: Add xmr and dcr adaptor signature primitives.
https://proposals.decred.org/record/fa0ea64

We need the script logic for xmr and dcr adaptor signatures. Some resources:
[technical paper](https://eprint.iacr.org/2020/1126.pdf)
[basicswap github](https://github.com/tecnovert/basicswap)

### PR DESCRIPTION
**Pull request by JoeGruffins**
closes #2754

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: ['1.20', '1.21']
```
Comment (JoeGruffins) [inline .github/workflows/build.yml:11]: At least of of the imports requires 1.21 so the github testing suite will error without this change.

File: internal/cmd/xmrswap/main.go [hunk 2]
```diff
@@ -0,0 +1,1221 @@
...
+
+type client struct {
+	xmr *rpc.Client
+	dcr *combinedClient
+
+	kbvf, kbsf, kbvl, kbsl, vkbv *edwards.PrivateKey
+	pkbsf, pkbs                  *edwards.PublicKey
+	kaf, kal                     *secp256k1.PrivateKey
+	pkal, pkaf, pkasl, pkbsl     *secp256k1.PublicKey
+	kbsfDleag, kbslDleag         [libsecp256k1.ProofLen]byte
```
Comment (martonp) [inline internal/cmd/xmrswap/main.go:91]: It's really hard to follow the code with these variable names. Something like `ourSpendKey`, `counterPartySpendKey`, etc. would make it way more understandable.

Comment (JoeGruffins) [reply internal/cmd/xmrswap/main.go:91]: Better clean these up on top of #2936? No reason to create conflicts there imo.

File: internal/cmd/xmrswap/main.go [hunk 3]
```diff
@@ -0,0 +1,1221 @@
...
+	if err != nil {
+		return fail(err)
+	}
+
+	txBytes, err = hex.DecodeString(fundRes.Hex)
+	if err != nil {
+		return fail(err)
+	}
+
+	c.lockTx = wire.NewMsgTx()
```
[the remark this answers is not in the document]
Comment (JoeGruffins) [reply internal/cmd/xmrswap/main.go:449]: Renamed the top one. https://github.com/decred/dcrdex/compare/fc1f63d0517f126920cdfbb82140ab8258d07d00..d6b0917a5541b9c1fb509ebdf4f4d46123a21b31

File: dex/testing/loadbot/go.mod [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module decred.org/dcrdex/dex/testing/loadbot
 
-go 1.21
+go 1.21.10
+
+toolchain go1.23.0
```
Comment (JoeGruffins) [inline dex/testing/loadbot/go.mod:5]: My `go mod tidy` seems to be adding the toolchain here. We don't want that though, correct?

Comment (JoeGruffins) [reply dex/testing/loadbot/go.mod:5]: I guess one of the new imports is forcing this.

Comment (ukane-philemon) [reply dex/testing/loadbot/go.mod:5]: You can confirm if you can build without it.

Comment (JoeGruffins) [reply dex/testing/loadbot/go.mod]: I think it's fixed. It was @dev-warrior777 import but they fixed it.

File: internal/cmd/xmrswap/main.go [hunk 5]
```diff
@@ -0,0 +1,1221 @@
...
+	"github.com/decred/dcrd/txscript/v4/sign"
+	"github.com/decred/dcrd/txscript/v4/stdaddr"
+	"github.com/decred/dcrd/wire"
+	"github.com/decred/slog"
+	"github.com/dev-warrior777/go-monero/rpc"
+	"github.com/fatih/color"
+	"github.com/haven-protocol-org/monero-go-utils/base58"
+)
+
+// TODO: Verification at all stages has not been implemented yet.
```
Comment (dev-warrior777) [inline internal/cmd/xmrswap/main.go:41]: This TODO still valid?

**Review by dev-warrior777 — APPROVED** [review]
All good!

File: internal/cmd/xmrswap/main.go (hunk 5 above)
Comment (JoeGruffins) [reply internal/cmd/xmrswap/main.go:41]: Yes, we are not inspecting the on-chain txn at all yet.

**Comment by JoeGruffins** [comment]
I guess this was supposed to go in before marton's pr, my miss. These changes are already in master.
